hubblestack/splunklogging.py

```diff
@@ -0,0 +1,315 @@
+        log_entry = self.format_record(record)
+        for hec, event, payload in self.endpoint_list:
+            event = copy.deepcopy(event)
+            payload = copy.deepcopy(payload)
+            event.update(log_entry)
+            payload['event'] = event
+            hec.batchEvent(payload, eventtime=time.time())
+            hec.flushBatch()
+        return True
+
+    def format_record(self, record):
+        '''
+        Format the log record into a dictionary for easy insertion into a
+        splunk event dictionary
+        '''
+        log_entry = {'message': record.message,
+                     'level': record.levelname,
+                     'timestamp': record.asctime,
+                     'loggername': record.name,
+                     }
+        return log_entry
+
+
+def _get_options():
+    if __salt__['config.get']('hubblestack:returner:splunk'):
+        splunk_opts = []
+        returner_opts = __salt__['config.get']('hubblestack:returner:splunk')
+        if not isinstance(returner_opts, list):
+            returner_opts = [returner_opts]
+        for opt in returner_opts:
+            processed = {}
+            processed['token'] = opt.get('token')
+            processed['indexer'] = opt.get('indexer')
+            processed['port'] = str(opt.get('port', '8088'))
+            processed['index'] = opt.get('index')
+            processed['custom_fields'] = opt.get('custom_fields', [])
+            processed['sourcetype'] = opt.get('sourcetype_log', 'hubble_log')
+            processed['http_event_server_ssl'] = opt.get('hec_ssl', True)
+            processed['proxy'] = opt.get('proxy', {})
+            processed['timeout'] = opt.get('timeout', 9.05)
+            processed['index_extracted_fields'] = opt.get('index_extracted_fields', [])
+            splunk_opts.append(processed)
+        return splunk_opts
+    else:
+        splunk_opts = {}
+        splunk_opts['token'] = __salt__['config.get']('hubblestack:nova:returner:splunk:token').strip()
+        splunk_opts['indexer'] = __salt__['config.get']('hubblestack:nova:returner:splunk:indexer')
+        splunk_opts['port'] = __salt__['config.get']('hubblestack:nova:returner:splunk:port', '8088')
+        splunk_opts['index'] = __salt__['config.get']('hubblestack:nova:returner:splunk:index')
+        splunk_opts['custom_fields'] = __salt__['config.get']('hubblestack:nova:returner:splunk:custom_fields', [])
+        splunk_opts['sourcetype'] = __salt__['config.get']('hubblestack:nova:returner:splunk:sourcetype')
+        splunk_opts['http_event_server_ssl'] = __salt__['config.get']('hubblestack:nova:returner:splunk:hec_ssl', True)
+        splunk_opts['proxy'] = __salt__['config.get']('hubblestack:nova:returner:splunk:proxy', {})
+        splunk_opts['timeout'] = __salt__['config.get']('hubblestack:nova:returner:splunk:timeout', 9.05)
+        splunk_opts['index_extracted_fields'] = __salt__['config.get']('hubblestack:nova:returner:splunk:index_extracted_fields', [])
```

Comment:
Looks like the old style config is pointing to the nova config here.

Reply:
This is actually on purpose. I didn't want to add a new block of old style config (since I'd like to deprecate that config eventually), and nova was the most likely block to exist.

That said, I think I'm going to require new style config for this one. Since it has to be opt in anyway, they should be updating their config style.